particle/webpack.md

```diff
@@ -1,7 +1,33 @@
+
+**App specific config**: Each application contains it's own webpack configuration to specify settings needed to run it. This may include dev server settings, overrides to previously established file loaders, as well as the entry and output points for the app. Each app config has a webpack object for `shared`, `dev`, and `prod`. If there are no specific settings for one of these environments the variable is an empty object. Each app configuration exports a call to the particle function from `particle.js` to create the webpack configuration object for that application.
+
+## Understanding CSS Modes
+
+The particle function exported from `particle.js` also accepts a `cssMode` key on the options object. The cssMode can be set to `'hot'` or `'extract'` to alter how Webpack handles the CSS assets. If no key is passed, particle will default to extract mode. Hot mode loads the styles within the JavaScript bundle in memory and allows for hot reloading to be utilized. It is meant for fast development experiences, but should not be used in production. Extract mode emits the CSS assets as CSS files on disk. This is desirable for production environments so that JavaScript does not need to be parsed and executed to load the stylesheets. The CSS Modes option is set in each app specific config.
+
+## JavaScript polyfill
+
+Babel transpiles JavaScript during the build process to convert from newer ES6+ syntax to ES5 syntax that browsers can process. Particle also ships with the Babel polyfill on production builds. This covers changes to the actual language which may not exist in older browsers. For example, the `fetch` protocol is a relatively recent addition and does not exist on certain versions of IE. In order for the code to function properly a polyfill must be used to bridge that functionality. Babel's polyfill reads from the `.browserslistrc` file to only polyfill the features that are needed for the targeted browsers.
+
+## Bundle Warnings during build
+
+The default Particle install will emit Webpack warnings when running the `build:pl` script. These warnings are: `WARNING in asset size limit`, and `WARNING in entrypoint size limit`. The asset size limit is a result of the Font Awesome SVG map. This raw asset is tree-shaken during the build process to only utilize the icons used and referenced in the code base. The entrypoint size limit is a result of the decision to bundle everything into one entrypoint for initial install and ease of use. This can be optimized through creating multiple entry points, or code-splitting the bundle as described in the section below. Additionally, if you don't need libraries such as Vue for your application, the bundle size can be reduced by removing Vue examples from the dependency chain. These examples are located at `source/default/_patterns/02-molecules/vue-widget/`.
```

Comment:
Not necessarily a part of this PR, but I think this is a good reason to pull Vue/TypeScript off of the Carousel example. We get a lot of basic Drupal usage out of the Carousel example, and advocating it be removed in Sprint 0 causes additional questions.

tl;dr: I've answered quite a few usage questions in the past month that could be solved by making Carousel a good Drupal usage example.

Reply:
Originally the Carousel was suggested by @illepic to be converted over due to the settings object being a good opportunity to show the interface. I'm not opposed to converting it back as part of the PR either as we definitely don't want to make the example usage get in the way.

Reply:
For sure! I thought about not saying anything. But imo, it probably would be better to make a decision about our demo components. I think we should use each individually to demonstrate a given concept. That way, when there's a question, we can succinctly say, "check this out here."

Carousel is a perfect example to demonstrate how to pass settings to Drupal.